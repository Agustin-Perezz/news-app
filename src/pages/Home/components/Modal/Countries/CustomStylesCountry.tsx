import { styled } from '@mui/material/styles';
import MuiBox from '@mui/material/Box';

export const CustomCountryBox = styled(MuiBox)(() => ({
  padding: '3px 4px',
  margin: '5px 0px 5px 5px',
  border: '1px solid #45454b',
  borderRadius: '5px',
  '&:hover': {
    cursor: 'pointer',
    border: '1px solid #8a8a90',
  },
}));
