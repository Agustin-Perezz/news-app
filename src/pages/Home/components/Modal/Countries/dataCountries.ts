export const dataCountries = [
  { code: 'ar', label: 'Argentina' },
  { code: 'au', label: 'Australia' },
  { code: 'us', label: 'United States Of America' },
  { code: 'br', label: 'Brazil' },
  { code: 'bo', label: 'Bolivia' },
  { code: 'ca', label: 'Canada' },
  { code: 'at', label: 'Austria' },
  { code: 'cl', label: 'Chile' },
  { code: 'cn', label: 'China' },
  { code: 'es', label: 'Espa√±a' },
  { code: 'ua', label: 'Ukraine' },
  { code: 'co', label: 'Colombia' },
  { code: 'hr', label: 'Croatia' },
  { code: 'ec', label: 'Ecuador' },
  { code: 'fr', label: 'France' },
  { code: 'de', label: 'Germany' },
  { code: 'hn', label: 'Honduras' },
  { code: 'in', label: 'India' },
  { code: 'ir', label: 'Iran' },
  { code: 'it', label: 'Italy' },
  { code: 'jp', label: 'Japan' },
  { code: 'mx', label: 'Mexico' },
  { code: 'kr', label: 'Korea' },
  { code: 'pe', label: 'Peru' },
  { code: 'pt', label: 'Portugal' },
  { code: 'uy', label: 'Uruguay' },
];
