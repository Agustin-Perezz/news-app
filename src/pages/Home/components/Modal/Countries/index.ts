export * from './ListCountries';
export * from './dataCountries';
