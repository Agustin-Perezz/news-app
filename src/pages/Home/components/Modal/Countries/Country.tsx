import { Tooltip } from '@mui/material';
import { HandleClickCountryProps } from '../../../types';
import { CustomCountryBox } from './CustomStylesCountry';

type Props = HandleClickCountryProps & {
  code: string;
  title: string;
};

export const Country: React.FC<Props> = ({ code, title, handleClickCountry }) => {
  return (
    <CustomCountryBox onClick={() => handleClickCountry(code)}>
      <Tooltip title={title} arrow>
        <img
          src={`https://flagcdn.com/36x27/${code}.png`}
          alt={title}
          style={{ verticalAlign: 'middle' }}
        />
      </Tooltip>
    </CustomCountryBox>
  );
};
