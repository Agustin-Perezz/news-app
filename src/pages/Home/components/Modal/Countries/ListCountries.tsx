import { Box } from '@mui/material';
import { Country } from './Country';
import { dataCountries } from './dataCountries';

export const ListCountries: React.FC = () => {
  return (
    <Box sx={{ display: 'flex', justifyContent: 'center', flexWrap: 'wrap' }}>
      {dataCountries.map(({ code, label }) => (
        <Country code={code} title={label} key={label} />
      ))}
    </Box>
  );
};
