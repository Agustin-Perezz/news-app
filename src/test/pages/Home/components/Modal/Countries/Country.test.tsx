import { fireEvent, render, screen } from '@testing-library/react';
import { Country } from '../../../../../../pages/Home/components/Modal/Countries/Country';

describe('Test in <Country.test />', () => {
  const handleClickCountry = jest.fn();
  function setup() {
    return render(<Country code="ar" title="Argentina" handleClickCountry={handleClickCountry} />);
  }

  it('should match snapshot', () => {
    setup();
    expect(screen).toMatchSnapshot();
  });
  it('should render props correctly', () => {
    setup();
    const countryElement = screen.getByRole('img');
    expect(countryElement).toBeInTheDocument();
    expect(countryElement).toHaveAttribute('aria-label', 'Argentina');
    expect(countryElement).toHaveAttribute('src', 'https://flagcdn.com/36x27/ar.png');
  });
  it('should call handleClickCountry', () => {
    const { container } = setup();
    const countryBox = container.firstElementChild;
    fireEvent.click(countryBox!);
    expect(handleClickCountry).toHaveBeenCalledWith('ar', 'Argentina');
  });
});
