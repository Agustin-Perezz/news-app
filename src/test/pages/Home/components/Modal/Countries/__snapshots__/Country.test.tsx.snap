// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Country /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1aqkl79"
  >
    <img
      alt="Argentina"
      aria-label="Argentina"
      class=""
      data-mui-internal-clone-element="true"
      src="https://flagcdn.com/36x27/ar.png"
      style="vertical-align: middle;"
    />
  </div>
</div>
`;
